src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/arrivals/ConfirmationService.kt

```diff
@@ -89,84 +83,43 @@ class ConfirmationService(
     }
 
   private fun admitOffenderOnNewBooking(confirmation: Confirmation, prisonNumber: String): ConfirmArrivalResponse {
-    val response = prisonService.admitOffenderOnNewBooking(prisonNumber, confirmation.detail)
-
-    whenIsExpectedArrival(confirmation) { arrivalId, detail ->
-      confirmedArrivalRepository.save(
-        ConfirmedArrival(
-          movementId = arrivalId,
-          prisonNumber = prisonNumber,
-          timestamp = LocalDateTime.now(clock),
-          arrivalType = ArrivalType.NEW_BOOKING_EXISTING_OFFENDER,
-          prisonId = detail.prisonId!!,
-          bookingId = response.bookingId,
-          arrivalDate = LocalDate.now(clock),
-          username = securityUserContext.principal,
-        )
-      )
-    }
-    return createResponse(prisonNumber, response)
+    val result = prisonService.admitOffenderOnNewBooking(prisonNumber, confirmation.detail)
+
+    arrivalListener.arrived(confirmation.toEvent(result, NEW_BOOKING_EXISTING_OFFENDER))
```

Comment:
put a layer of abstraction over the confirmed arrival repository. 

As well as improving readability of these methods, this meant I could remove the clock and security user context dependencies out of here simplifying the tests.

The mean reason though was to be able to add a shim to adding publishing app insights events into a single place rather than each method.